Library/Homebrew/cask/cask_loader.rb

```diff
@@ -199,16 +199,22 @@ class FromTapLoader < FromPathLoader
       attr_reader :tap
 
       sig {
-        params(ref: T.any(String, Pathname, Cask, URI::Generic), warn: T::Boolean)
-          .returns(T.nilable(T.attached_class))
+        override(allow_incompatible: true) # rubocop:todo Sorbet/AllowIncompatibleOverride
+          .params(ref: T.any(String, Pathname, Cask, URI::Generic), warn: T::Boolean)
+          .returns(T.nilable(T.any(T.attached_class, FromAPILoader)))
```

Comment:
This is the hack to allow Sorbet to change the return type here. We can't use the common `Cask::CaskLoader::ILoader` interface because it doesn't have some needed methods defined like `#path`.